data/dataexchange/src/main/java/tech/pegasys/teku/data/SlashingProtectionImporter.java

```diff
@@ -129,35 +129,56 @@ private SigningHistory signingHistoryConverter(final SigningHistory signingHisto
     }
   }
 
-  public void updateLocalRecords(final Path slashingProtectionPath) {
+  public void updateLocalRecords(
+      final Path slashingProtectionPath, final Consumer<String> statusConsumer) {
     this.slashingProtectionPath = slashingProtectionPath;
-    data.forEach(this::updateLocalRecord);
-    log.display("Updated " + data.size() + " validator slashing protection records");
+    data.forEach(record -> updateLocalRecord(record, statusConsumer));
+    statusConsumer.accept("Updated " + data.size() + " validator slashing protection records");
   }
 
-  private void updateLocalRecord(final SigningHistory signingHistory) {
+  public Optional<String> updateSigningRecord(final BLSPublicKey publicKey) {
+    final BLSPubKey key = new BLSPubKey(publicKey);
+    data.stream()
+        .filter(signingHistory -> signingHistory.pubkey.equals(key))
+        .collect(Collectors.toList())
+        .forEach(record -> updateLocalRecord(record, (__) -> {}));
+    if (validatorImportErrors.containsKey(publicKey)) {
+      return Optional.of(validatorImportErrors.get(publicKey));
+    }
+    return Optional.empty();
+  }
+
+  private void updateLocalRecord(
```

Comment:
Would it make more sense to have this return `Optional<String>` rather than putting errors into the `validatorImportErrors` map?

The public `updateLocalRecords` method could then locally collect any errors into a map to return rather than needing a separate `getValidatorImportErrors` method.